core/src/plugins/filed/python/vmware/bareos-fd-vmware.py

```diff
@@ -2957,35 +2957,47 @@ def cleanup_tmp_files(self):
 
     def fetch_vcthumbprint(self):
         """
-        Retrieve the SSL Cert thumbprint from VC Server
+        Fetch the SSL Cert thumbprint from VC Server
         """
         if self.fetched_vcthumbprint:
             return True
 
-        success = True
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        sock.settimeout(5)
-        wrappedSocket = ssl.wrap_socket(sock)
+        success = False
+        error_message = ""
+        ssl_context = ssl.create_default_context()
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+
         bareosfd.DebugMessage(
             100,
-            "retrieve_vcthumbprint() Retrieving SSL ThumbPrint from %s\n"
+            "fetch_vcthumbprint() Retrieving SSL ThumbPrint from %s\n"
             % (self.options["vcserver"]),
         )
+
         try:
-            wrappedSocket.connect((self.options["vcserver"], 443))
+            sock = socket.create_connection((self.options["vcserver"], 443), timeout=5)
```

Comment:
this looks way more complicated than it actually is.
As we're not handling exceptions, but just reporting that they happened, we could just reduce everything to one single try/except block like this:
```
        try:
            with socket.create_connection((self.options["vcserver"], 443), timeout=5) as sock:
                with ssl_context.wrap_socket(
                    sock, server_hostname=self.options["vcserver"]
                ) as wrappedSocket:
                    der_cert_bin = wrappedSocket.getpeercert(True)
                    thumb_sha1 = hashlib.sha1(der_cert_bin).hexdigest()
                    self.fetched_vcthumbprint = thumb_sha1.upper()
            return True
        except Exception as e:
            bareosfd.JobMessage(
                bareosfd.M_FATAL,
               "Could not fetch SSL Cert from %s: %s\n"
                % (self.options["vcserver"], str(e)),
            )
            return False
```

Reply:
Sure, that looks a lot better indeed. Thanks.